### DISCUSSION TITLE
Styling Alerts

### DISCUSSION THREAD

**Discussion by shaneajeffery** [chakra-ui/chakra-ui#4237] [Q&A]
Hello all.

I am needing to override some of the styling for Alerts based upon whatever `status` is getting passed in.  Given that `status` controls a lot of children pieces of Alerts, I had to get creative and just ended up doing an `styled` wrapper around the `Alert` component.  With that though, I can't get access to my top-level Chakra themeing.

Is there a way to get this working the Chakra-way so that I can access my theme variables?

This is the code that I am currently doing:

`styles.tsx`

```js
import { Alert } from '@chakra-ui/react';
import { css } from '@emotion/react';
import styled from '@emotion/styled';

const buildStyles = (props: any) => {
  let styles;

  switch (props.status) {
    case 'info':
      styles = css`
        background: #f2f6fa;
        box-sizing: border-box;
        border: 1px solid #3d4858;
        border-radius: 8px;
        color: #1e394e;
      `;
      break;
    case 'success':
      styles = css`
        background: #e7f4e4;
        box-sizing: border-box;
        border: 1px solid #48a463;
        border-radius: 8px;
        color: #1e394e;
      `;
      break;
    case 'error':
      styles = css`
        background: #f9e9e8;
        box-sizing: border-box;
        border: 1px solid #d46a67;
        border-radius: 8px;
        color: #1e394e;
      `;
      break;
  }

  return styles;
};

export const CKAlert = styled(Alert)`
  ${buildStyles};
`;

```

`index.tsx`

```js
import { AlertIcon, AlertDescription, CloseButton } from '@chakra-ui/react';
import React from 'react';

import { CKAlert } from './styles';

type TAlertStatus = 'info' | 'success' | 'error';

interface Props {
  status: TAlertStatus;
  children: any;
}

export const OntadaAlert = (props: Props) => {
  return (
    <CKAlert status={props.status}>
      <AlertIcon />
      <AlertDescription>{props.children}</AlertDescription>
      <CloseButton position="absolute" right="8px" top="8px" />
    </CKAlert>
  );
};
```

I have tried to use the `chakra` factory, but I cannot figure out how to get the `props` out of that factory like I can with just regular styled components.

Any help would be appreciated!  Thanks!

**Comment by anubra266** [comment]
I don't really get what you mean by accessing your theme variables. But check out [useTheme](https://chakra-ui.com/docs/hooks/use-theme#import)

**Reply by shaneajeffery** [reply]
I mean, that I have created a theme that is use by the `ChakraProvider`.  In it, I have extended the theme to allow for more colors.  Normally, I have access to those colors by just doing 'themeColors.darkSlate'.

Given that my emotion styled components aren't under Chakra though, I can't get to those new colors.

The `useTheme` hook is only available within a React class or FC and not available in styled components.

**Reply by anubra266** [reply]
Well, the above way is not the standard way to customize Chakra components. 

Though judging by what you're doing above. You should know you can change the scheme of an Alert.
![image](https://user-images.githubusercontent.com/30869823/122301748-06ba8e00-cef9-11eb-8cc2-8a0a98f57f63.png)
 
Now in trying to change border and all, check out [Styling Multipart Components](https://chakra-ui.com/docs/theming/component-style#styling-multipart-components)

Alert's default theme is [here](https://github.com/chakra-ui/chakra-ui/blob/main/packages/theme/src/components/alert.ts)
found at the top of the `Alert` component page in docs.
![image](https://user-images.githubusercontent.com/30869823/122302168-a4ae5880-cef9-11eb-806e-40142db6dacb.png)
